src/hooks/useWeb3React.tsx

```diff
@@ -81,3 +84,30 @@ function getLibrary(provider: any): Web3Provider {
   library.pollingInterval = 12000;
   return library;
 }
+
+const Web3EventHooks = () => {
+  useEffect(() => {
+    const ethereum = (window as any).ethereum;
+    if (ethereum) {
+      // The "any" network will allow spontaneous network changes
+      const provider = new Web3Provider(ethereum, 'any');
+
+      provider.on('network', (_, oldNetwork) => {
+        // When a Provider makes its initial connection, it emits a "network"
+        // event with a null oldNetwork along with the newNetwork. So, if the
+        // oldNetwork exists, it represents a changing network
+        if (oldNetwork) {
+          window.location.reload();
+        }
+      });
+
+      // Web3Provider doesn't work with accountsChanged
+      ethereum.on('accountsChanged', () => {
```

Comment:
@topocount not sure why but using `provider.on` doesn't work here. Any idea why the Ethers provider wouldn't pass through this event? But using the plain object does work?

Reply:
@crabsinger I'm not sure what

> using the plain object does work

means. Could you sketch the working example?

Reply:
`ethereum.on('accountsChanged', ()` works
`provider.on('network',` does not work.

@topocount does that make sense? Wrapping the `window.ethereum` object in a Ethers Web3Provider breaks the functionality.

Reply:
https://github.com/ethers-io/ethers.js/issues/1396#issuecomment-806380431 answer here!